src/main/java/com/amazon/dlic/auth/ldap/util/LdapHelper.java

```diff
@@ -40,30 +40,40 @@ public class LdapHelper {
     private static SearchFilter ALL = new SearchFilter("(objectClass=*)");
     @SuppressWarnings("removal")
     public static List<LdapEntry> search(final Connection conn, final String unescapedDn, SearchFilter filter,
-            final SearchScope searchScope, final String[] returnAttributes) throws LdapException {
+            final SearchScope searchScope, final String[] returnAttributes, boolean shouldFollowReferrals) throws LdapException {
 
         final SecurityManager sm = System.getSecurityManager();
 
         if (sm != null) {
             sm.checkPermission(new SpecialPermission());
         }
-
         try {
             final String baseDn = escapeDn(unescapedDn);
             return AccessController.doPrivileged(new PrivilegedExceptionAction<List<LdapEntry>>() {
                 @Override
                 public List<LdapEntry> run() throws Exception {
                     final List<LdapEntry> entries = new ArrayList<>();
                     final SearchRequest request = new SearchRequest(baseDn, filter);
-                    request.setReferralHandler(new SearchReferralHandler());
+
                     request.setSearchScope(searchScope);
                     request.setDerefAliases(DerefAliases.ALWAYS);
                     request.setReturnAttributes(returnAttributes);
                     final SearchOperation search = new SearchOperation(conn);
-                    // referrals will be followed to build the response
+
+                    if (shouldFollowReferrals) {
+                         // referrals will be followed to build the response
+                        request.setReferralHandler(new SearchReferralHandler());
+                    }
+
                     final Response<SearchResult> r = search.execute(request);
                     final org.ldaptive.SearchResult result = r.getResult();
                     entries.addAll(result.getEntries());
+
+                    if (entries == null) {
```

Comment:
This cannot be null, its value is set at 55, this check/exception should be removed

Reply:
Fixed!